packages/core/src/App/Containers/RealAccountSignup/address-details.jsx

```diff
@@ -37,206 +37,170 @@ const InputField = props => {
     );
 };
 
-class AddressDetails extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            has_fetched_states_list: false,
-            address_state_to_display: '',
-        };
-        // TODO: Find a better solution for handling no-op instead of using is_mounted flags
-        this.is_mounted = false;
-    }
-
-    async componentDidMount() {
-        this.is_mounted = true;
-        await this.props.fetchStatesList();
-        if (this.is_mounted)
-            this.setState({
-                has_fetched_states_list: true,
-                address_state_to_display: getLocation(this.props.states_list, this.props.value.address_state, 'text'),
-            });
-    }
+const AddressDetails = props => {
+    const [has_fetched_states_list, setHasFetchedStatesList] = React.useState(false);
+    const [address_state_to_display, setAddressStateToDisplay] = React.useState('');
 
-    componentWillUnmount() {
-        this.is_mounted = false;
-    }
+    React.useEffect(() => {
+        const fetchedStateList = async () => {
+            await props.fetchStatesList();
+            setHasFetchedStatesList(true);
+            setAddressStateToDisplay(getLocation(props.states_list, props.value.address_state, 'text'));
```

Comment:
How about destructuring required variables first instead of calling them directly on `props`?

Reply:
In fact, I tried to minimise the changes on the previous code, but alright, i will update the code.